From fd3cf6d33d67232266d0d8df401a0ded78a95c56 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@freescale.com>
Date: Thu, 18 Oct 2012 15:44:29 -0500
Subject: [PATCH 077/310] ENGR00230124 - Need to add audio HAL for JB

Adding mixer_paths.xml to PRODUCT_COPY_FILES.
Changing audio_policy.conf to use PRODUCT_COPY_FILES.

Signed-off-by: Oliver Brown <oliver.brown@freescale.com>
---
 imx53_smd/audio_policy.conf | 41 +++++++++++++++++++++++++++++++++++++++++
 imx53_smd/mixer_paths.xml   | 18 ++++++++++++++++++
 imx5x/imx53_smd.mk          |  5 +++++
 imx5x/imx5x.mk              |  1 -
 4 files changed, 64 insertions(+), 1 deletion(-)
 create mode 100644 imx53_smd/audio_policy.conf
 create mode 100644 imx53_smd/mixer_paths.xml

diff --git a/imx53_smd/audio_policy.conf b/imx53_smd/audio_policy.conf
new file mode 100644
index 0000000..e350504
--- /dev/null
+++ b/imx53_smd/audio_policy.conf
@@ -0,0 +1,41 @@
+# Global configuration section: lists input and output devices always present on the device
+# as well as the output device selected by default.
+# Devices are designated by a string that corresponds to the enum in audio.h
+
+global_configuration {
+  attached_output_devices AUDIO_DEVICE_OUT_EARPIECE|AUDIO_DEVICE_OUT_SPEAKER
+  default_output_device AUDIO_DEVICE_OUT_SPEAKER
+  attached_input_devices AUDIO_DEVICE_IN_BUILTIN_MIC
+}
+
+# audio hardware module section: contains descriptors for all audio hw modules present on the
+# device. Each hw module node is named after the corresponding hw module library base name.
+# For instance, "primary" corresponds to audio.primary.<device>.so.
+# The "primary" module is mandatory and must include at least one output with
+# AUDIO_OUTPUT_FLAG_PRIMARY flag.
+# Each module descriptor contains one or more output profile descriptors and zero or more
+# input profile descriptors. Each profile lists all the parameters supported by a given output
+# or input stream category.
+# The "channel_masks", "formats", "devices" and "flags" are specified using strings corresponding
+# to enums in audio.h and audio_policy.h. They are concatenated by use of "|" without space or "\n".
+audio_hw_modules {
+  primary {
+    outputs {
+      primary {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_EARPIECE|AUDIO_DEVICE_OUT_SPEAKER|AUDIO_DEVICE_OUT_WIRED_HEADSET|AUDIO_DEVICE_OUT_WIRED_HEADPHONE|AUDIO_DEVICE_OUT_ALL_SCO|AUDIO_DEVICE_OUT_AUX_DIGITAL|AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET
+        flags AUDIO_OUTPUT_FLAG_PRIMARY
+      }
+    }
+    inputs {
+      primary {
+        sampling_rates 8000|16000
+        channel_masks AUDIO_CHANNEL_IN_MONO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_IN_BUILTIN_MIC
+      }
+    }
+  }
+}
diff --git a/imx53_smd/mixer_paths.xml b/imx53_smd/mixer_paths.xml
new file mode 100644
index 0000000..4a8cc57
--- /dev/null
+++ b/imx53_smd/mixer_paths.xml
@@ -0,0 +1,18 @@
+<mixer>
+  <!-- These are the initial mixer settings -->
+    <ctl name="Speaker Function" value="on" />
+    <ctl name="Jack Function" value="on" />
+  <ctl name="PCM Playback Volume" value="160" />
+  <path name="speaker">
+      <ctl name="Speaker Function" value="on" />
+      <ctl name="PCM Playback Volume" value="160" />
+  </path> 
+  <path name="main-mic">
+      <ctl name="Speaker Function" value="on" />
+      <ctl name="PCM Playback Volume" value="160" />
+  </path> 
+  <path name="headphone">
+      <ctl name="Speaker Function" value="on" />
+      <ctl name="PCM Playback Volume" value="160" />
+  </path> 
+</mixer>
diff --git a/imx5x/imx53_smd.mk b/imx5x/imx53_smd.mk
index d1acd4a..00ae463 100644
--- a/imx5x/imx53_smd.mk
+++ b/imx5x/imx53_smd.mk
@@ -37,6 +37,11 @@ PRODUCT_COPY_FILES += \
     packages/wallpapers/LivePicker/android.software.live_wallpaper.xml:system/etc/permissions/android.software.live_wallpaper.xml
 endif
 
+# audio related files
+PRODUCT_COPY_FILES += \
+	device/fsl/imx53_smd/audio_policy.conf:system/etc/audio_policy.conf \
+	device/fsl/imx53_smd/mixer_paths.xml:system/etc/mixer_paths.xml
+
 PRODUCT_PREF_AAPT_CONFIG := mdpi
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/imx53_smd/overlay
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index 671108f..721cfc1 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -59,7 +59,6 @@ PRODUCT_PACKAGES += \
 #audio related lib
 PRODUCT_PACKAGES += \
 	audio.primary.imx5			\
-	audio_policy.conf			\
 	tinyplay				\
 	tinycap					\
 	tinymix					\
-- 
1.8.0

